### PR TITLE
Cherry pick changes from point-release/23101 to development

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
## What does this PR do?

Generated the list of commits to cherry-pick using 
```bash
git log --reverse --format=%H --first-parent upstream/development..upstream/point-release/23101 --since="Thu Oct 19 08:01:42 2023 -0700" --cherry-pick
```
manually went through commits and expanded merge commits to actual commits as needed.

CONFLICT commits:
a923f3cae58043dc28e71f0bf414f83f86b1aaac @gadams3 this one has to do with the defaultSamplerState variable
4fec5cfd617e6daecca6ef521b9fcf5dba725161 @spham-amzn HEADLESS vs HEADLESS_SERVER
dc6e2f7a2396aa1b9722bdae1c5453d7727266f9 @spham-amzn  HEADLESS
c4becde21ba3e75a42714ef2391614a805b63c20 @spham-amzn @tkothadev export
90ea3702767de2530fb1f21457bf86d29c59e9bd @alexmontAmazon 
f2654818bf3919d3e0ac77c54e199c73ae5d3c03 @tkothadev export
851b7d8a322c84651971959582081d99c7b602bf @gadams3 Culling optimizations
5777f25bd89ad8e3e68e6897b03bdda7e9724539 @AMZN-alexpete generator removal
6aa78b5d9fd29ead1a8eff261268e4f8264085c7 @spham-amzn ExampleLevel vs DefaultLevel in templates

## How was this PR tested?

Configured and built the Editor in profile on Windows with VS 2022

### CODE REVIEW

**Review by tkothadev — APPROVED** [review]
Project export cherry pick changes look good to me

**Review by gadams3 — APPROVED** [review]
I need to compare to see if there were any obvious conflicts but I don't see anything here

**Review by alexmontAmazon — APPROVED** [review]
you and the teenagers have this down!

File: Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp [hunk 1]
```diff
@@ -497,7 +497,7 @@ namespace AZ
 
 #if AZ_TRAIT_MASKED_OCCLUSION_CULLING_SUPPORTED
             MaskedOcclusionCulling* maskedOcclusionCulling = worklistData->m_view->GetMaskedOcclusionCulling();
-            if (!maskedOcclusionCulling)
+            if (!maskedOcclusionCulling || || !worklistData->m_view->GetMaskedOcclusionCullingDirty())
```
[the remark this answers is not in the document]
Comment (AMZN-alexpete) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Culling.cpp:500]: ah that might have been a merge error on my part, good catch

File: scripts/o3de/o3de/export_project.py [hunk 2]
```diff
@@ -836,3 +1037,173 @@ def validate_project_artifact_paths(project_path: pathlib.Path,
...
+
+SETTINGS_OPTION_BUILD_SERVER_LAUNCHER = register_setting(key='option.build.server.launcher',
+                                                         description='The option to build an exported server launcher.',
+                                                         is_boolean=True,
+                                                         default='True')
+
+SETTINGS_OPTION_BUILD_HEADLESS_SERVER_LAUNCHER = register_setting(key='option.build.headless.server.launcher',
+                                                                  description='The option to build an exported headless server launcher.',
+                                                                  is_boolean=True,
+                                                                  default='False')
```
Comment (spham-amzn) [inline scripts/o3de/o3de/export_project.py:1180]: Very nice noticed and added this!

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**